### PR TITLE
Validate config item type and change child items to new struct

### PR DESCRIPTION
**Pull request by marccampbell**

### CODE REVIEW

File: config_child_item.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package libyaml
+
+// ConfigChildItem represents a subitem.  This is normally found in select_one and select_many types.
+type ConfigChildItem struct {
+	Name  string `yaml:"name" json:"name" validate:"required"`
+	Title string `yaml:"title" json:"title"`
```
Comment (emosbaugh) [inline config_child_item.go]: look at replicated-config-render we do use more than just this. add default, value, value_cmd, readonly, required, recommended

``` react
          <ConfigItemTitle
            title={item.title}
            required={item.required === true}
            recommended={item.recommended === true} />

            <ConfigFieldRadio
              title={title}
              name={item.name}
              group={this.props.name}
              value={this.props.value}
              default={this.props.default}
              readOnly={this.props.readOnly}
              handleChange={this.handleItemChange} />
```

Comment (marccampbell) [inline config_child_item.go]: The attributes that have values set to `this.props` are all getting values from the parent item.  Only the title and name are used in a child item.

Comment (emosbaugh) [inline config_child_item.go]: i do see this comment https://github.com/replicatedcom/replicated-config-render/blob/master/src/components/ConfigFieldSelectMany.js#L20

```
// TODO: does required make sense here?
```

**Comment by emosbaugh** [comment]
lgtm

**Pull request merged**